Review 'File System Access'. (#8169)
GoogleChrome/web.dev: src/site/content/en/blog/file-system-access/index.md

API to ensure that people can easily manage their files. See the
[security and permissions](#security-considerations) section for more information. {% endAside %}

## Current status {: #status }

<div>

| Step                                     | Status                |
| ---------------------------------------- | --------------------- |
| 1. Create explainer                      | [Complete][explainer] |
| 2. Create initial draft of specification | [Complete][spec]      |
| 3. Gather feedback & iterate on design   | [Complete][spec]      |
| 4. Origin trial                          | Complete              |
| **5. Launch**                            | **Complete**          |

> </div>

Reviewer: I think this table should remain. All our articles have it, even if all lines are marked as **complete**.

Reviewer: It’s mostly consistency with all other articles.

> flag](https://github.com/brave/brave-browser/issues/18979). Android support for
> Chrome is planned; you can track progress by starring
> [crbug.com/1011535](https://crbug.com/1011535).

Reviewer: ```suggestion
flag](https://github.com/brave/brave-browser/issues/18979). Android support is
not immediately planned, but you can track potential progress by starring
[crbug.com/1011535](https://crbug.com/1011535).
```

await file.move(directory, 'newer_name');
```

{% Aside 'warning' %} Due to some open questions regarding cross-file-system moves, `move()` was unavailable for folders and moves outside of the

> [origin private file system](#accessing-the-origin-private-file-system) before Chrome 102.{% endAside %}

Reviewer: This is still limited to OPFS. It's _not_ available generally.

### Accessing the origin private file system

The origin private file system is a storage endpoint that, as the name suggests, is private to the
origin of the page. While browsers typically implement this by persisting the contents of this
origin private file system to disk somewhere, it is _not_ intended that the contents be easily user
accessible. Similarly, there is _no_ expectation that files or directories with names matching the
names of children of the origin private file system exist. While the browser might make it seem that
there are files, internally—since this is an origin private file system—the browser might store

> these "files" in a database or any other data structure. Essentially, if you create with this API,
> do _not_ expect to find it matched one-to-one somewhere on the hard disk. You can operate as usual on

Reviewer: ```suggestion
these "files" in a database or any other data structure. Essentially, if you use this API,
do _not_ expect to find the created files matched one-to-one somewhere on the hard disk. You can operate as usual on
```
